include/rocksdb/utilities/table_properties_collectors.h

```diff
@@ -84,4 +85,99 @@ std::shared_ptr<CompactOnDeletionCollectorFactory>
 NewCompactOnDeletionCollectorFactory(size_t sliding_window_size,
                                      size_t deletion_trigger,
                                      double deletion_ratio = 0);
+
+// A factory of a table property collector that marks a SST file as
+// need-compaction when for the tiering use case, it observes, at least "D" data
+// entries that are already eligible to be placed on the last level but are not
+// yet on the last level. "D" is the compaction trigger threshold set via the
```

Comment:
I believe this would be D entries out of a whole SST file, which might vary in size or number of entries. Any reason not to use a fraction of entries as a threshold? That seems better to me. Unlike tombstones, it shouldn't matter how the eligible items are distributed within the SST file.